packages/mcp-client/src/Agent.ts

```diff
@@ -49,18 +49,28 @@ export class Agent extends McpClient {
 
 	constructor({
 		provider,
+		baseUrl,
 		model,
 		apiKey,
 		servers,
 		prompt,
-	}: {
-		provider: InferenceProvider;
+	}: (
+		| {
+				provider: InferenceProvider;
+				baseUrl?: undefined;
+		  }
+		| {
+				baseUrl: string;
+				provider?: undefined;
+		  }
+	) & {
 		model: string;
 		apiKey: string;
 		servers: StdioServerParameters[];
 		prompt?: string;
 	}) {
-		super({ provider, model, apiKey });
+		super(provider ? { provider, baseUrl, model, apiKey } : { provider, baseUrl, model, apiKey });
+		/// ^This shenanigan is just here to please an overzealous TS type-checker.
```

Comment:
🫠

Reply:
cc @coyotte508 if he has an idea to "fix" (in a later PR?)

Reply:
I already had to write something like this in https://github.com/huggingface/huggingface.js/pull/1322 I think (before the force-push)